### PR TITLE
[FEATURE][ML] Write data frame configuration to process

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/analytics/DataFrameAnalysis.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    public DataFrameAnalysis(String name) {
+        this.name = ExceptionsHelper.requireNonNull(name, NAME.getPreferredName());
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(NAME.getPreferredName(), name);
+        builder.endObject();
+        return null;
```
Comment (hendrikmuhs) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/analytics/DataFrameAnalysis.java]: return builder I guess?

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/analytics/process/AnalyticsBuilder.java [hunk 2]
```diff
@@ -34,10 +51,24 @@ public void build() throws IOException {
...
         return command;
     }
+
+    private void addConfigFile(List<String> command) throws IOException {
+        Path configFile = Files.createTempFile(env.tmpFile(), "analysis", ".conf");
+        filesToDelete.add(configFile);
+        try (OutputStreamWriter osw = new OutputStreamWriter(Files.newOutputStream(configFile),StandardCharsets.UTF_8);
+             XContentBuilder jsonBuilder = JsonXContent.contentBuilder()) {
+
+            config.toXContent(jsonBuilder, Payload.XContent.EMPTY_PARAMS);
```
Comment (hendrikmuhs) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/analytics/process/AnalyticsBuilder.java]: `ToXContent.EMPTY_PARAMS` removes the strange dep to watcher

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/analytics/process/AnalyticsProcessConfig.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(ROWS, rows);
+        builder.field(COLS, cols);
+        builder.field(MEMORY_LIMIT, memoryLimit.getBytes());
+        builder.field(THREADS, threads);
+        builder.field(ANALYSIS, analysis);
+        builder.endObject();
+        return null;
```
Comment (hendrikmuhs) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/analytics/process/AnalyticsProcessConfig.java]: return builder?

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**